### ISSUE TITLE
front-end not rendering/displaying content in live environment

### ISSUE THREAD

**Issue by jsidney** [terralist/terralist#183]
my team has just deployed the latest version of terralist to an AWS fargate environment and we are experiencing some very strange errors when trying to use the frontend... 
once a user is logged in, the only thing that can be seen is this error: (Something went wrong internally. Please contact the platform administrator.) . there are no error logs from the container and in fact every API call returns a 200. However, all of the XHR requests get cancelled. Could this be related to a CORS setting (or something similar)? Any advice would be hugely helpful as we are desperate to expose this private registry to our entire organisation (its just **that** cool!! 😉  )
![image](https://github.com/terralist/terralist/assets/7551131/3a22fe91-605c-4998-93a7-747b61a7db75)

**Comment by jsidney** [comment]
just to clarify....when running the entire system locally, we have no problems (ie: when everything is running off localhost etc). the reason why I'm leaning towards CORS related issues is because we now have a custom URL (and if something in the application is expecting all traffic to come from localhost, it would block the "cross-origin" - ie the new URL)

**Comment by valentindeaconu** [comment]
Hello! 

From the header, I can tell that you're running a dev build or you've built your own image from the source without specifying the build-time variables. 

If you're in the first case, that's not recommended and you should use an already build image or build your own.

If you're in the second case, I think you missed setting up the `TERRALIST_URL` environment variable, which should point to your custom domain. Check out the [configuration options](https://github.com/terralist/terralist/blob/master/docs/CONFIGURATION.md#options), by default, it will hold the `http://localhost:5758` value.

I did not configure (yet) a CORS configuration on Terralist and during my public test deployments (on Heroku and AWS) I did not encounter any issues with this. Let me know if that fixed your issue or if I should investigate further.

**Comment by jsidney** [comment]
@valentindeaconu  - thanks for the response. 
so the environment variables are all configured correctly (I didn't show the full setup, but I was pretty careful about that). 
i guess in order to run a production build, we would have to set the value in the docker file?
```
ARG VERSION="production"
```

**Comment by jsidney** [comment]
thanks for the heads up @valentindeaconu. 
Since I am nowhere near a golang expert, if I were to use the `task build -- release` approach, that would essentially take the place of:
```
RUN go build -a -v -o terralist \
    -ldflags="\
      -X 'main.Version=${VERSION}' \
      -X 'main.CommitHash=${COMMIT_HASH}' \
      -X 'main.BuildTimestamp=${BUILD_TIMESTAMP}' \
      -X 'main.Mode=release'" \
    ./cmd/terralist/main.go
```
in the dockerfile?

**Comment by valentindeaconu** [comment]
Yes, but that's the backend build. The task also takes care of building the frontend application (`yarn install` and `yarn build`).

If you want to run them manually, follow the task line by line. You can also use the release binaries from here the [v0.4.1 release page](https://github.com/terralist/terralist/releases/tag/v0.4.1) or you can pull an already built docker image from [the package repository](https://github.com/terralist/terralist/pkgs/container/terralist).

**Comment by jsidney** [comment]
@valentindeaconu  - i am trying to just download the release binaries (I used the linux_amd64 version)... and that 'works' but now all I get is:
![image](https://github.com/terralist/terralist/assets/7551131/07209b2f-b5a8-47e7-9930-89a4246f9822)
this is my dockerfile as it stands:
```
COPY docker-entrypoint.sh /usr/local/bin/
COPY config.yaml /etc/terralist/config.yaml
RUN chmod +x /usr/local/bin/docker-entrypoint.sh
#COPY --from=backend /go/src/terralist/terralist /usr/local/bin
RUN curl -L https://github.com/terralist/terralist/releases/download/v0.4.1/terralist_linux_amd64.zip -o terralist.zip
#RUN ls -la
RUN unzip terralist.zip && chmod +x terralist
RUN mv terralist /usr/local/bin/terralist

ENTRYPOINT [ "/usr/local/bin/docker-entrypoint.sh" ]
#ENTRYPOINT [ "docker-entrypoint.sh" ]
CMD [ "server", "--config=/etc/terralist/config.yaml" ]
```
I'm convinced that I am just overcomplicating the situation here, but I am seriously lost now 🤣

**Comment by jsidney** [comment]
i cant seem to find any prebuilt docker images either? are you sure that the link you provided is correct?

> If you want to run them manually, follow the task line by line. You can also use the release binaries from here the [v0.4.1 release page](https://github.com/terralist/terralist/releases/tag/v0.4.1) or you can pull an already built docker image from [the package repository](https://github.com/terralist/terralist/pkgs/container/terralist).

**Comment by jsidney** [comment]
@valentindeaconu - not sure if you are aware, but the docker images are private :) I am getting a 401 when I try to pull it.

**Comment by valentindeaconu** [comment]
Well, I wasn't aware of that. Thank you for pointing it out to me! During the migration to the organization, I must've forgotten to set it to public. 

Check it out now. It should be public.

**Comment by jsidney** [comment]
no problem at all! i figured that might have been the problem :)

**Comment by jsidney** [comment]
so i was able to launch the app from your docker image, but I am still getting the same error as before (internal platform error) after I log in. I've gone through all of the access logs (and application logs) that I can and I cannot seem to find the issue. the only thing that I have changed is that I added 2 lines of code to the entrypoint-script so that we can create the db connection string on startup.. but more than that nothing else is different. unless I have missed something really simple. (I'm going to share my config here)
```
url: https://terralist.tooling.synthesize.co.za
database-backend: postgresql
log-level: trace
token-signing-secret: secertsofstuff
oauth-provider: bitbucket
bb-workspace: synthesis_admin
cookie-secret: randomstringforcookies
modules-storage-resolver: s3
s3-bucket-region: eu-west-1
port: 5758
s3-bucket-name: xxxxxxx
postgres-url: xxxxxxx
bb-client-id: xxxxx
bb-client-secret: xxxxx
```
this is what I added to the entrypoint script:
```
dbhost="postgres://$TERRALIST_POSTGRES_USER:$TERRALIST_POSTGRES_PASSWORD@$TERRALIST_POSTGRES_HOST:5432/terralist"
echo $dbhost
echo -e "\npostgres-url: $dbhost\ns3-bucket-name: $TERRALIST_S3_BUCKET_NAME" >> /etc/terralist/config.yaml
```
and my docker file is:
```
FROM ghcr.io/terralist/terralist:v0.4.1

RUN apk add --no-cache \
      libcap~=2.66 \
      dumb-init~=1.2 \
      su-exec~=0.2 \
      curl \
      zip

COPY docker-entrypoint.sh /usr/local/bin/
COPY config.yaml /etc/terralist/config.yaml
RUN chmod +x /usr/local/bin/docker-entrypoint.sh
ENTRYPOINT [ "/usr/local/bin/docker-entrypoint.sh" ]
CMD [ "server", "--config=/etc/terralist/config.yaml" ]
```

**Comment by jsidney** [comment]
so.... i found the issue. its a weird one ill admit... but the timeouts set on the Axios API config for the front end were too short. 
so I upped the timeouts and I got it to work. 
i need to do some investigation to see why the response time is much longer on my environment but might be worth adding some sort of configurable timeout?

**Comment by mruz** [comment]
When I set custom `TERRALIST_URL` I'm also getting timeouts on `artifacts/` and `authorities/` endpoints. Can we set a timeout at runtime?

**Comment by l2dy** [comment]
@valentindeaconu The default timeout is 120ms, or 0.12 seconds. Did you mean to set it to 120 seconds instead?

**Comment by valentindeaconu** [comment]
> @valentindeaconu The default timeout is 120ms, or 0.12 seconds. Did you mean to set it to 120 seconds instead?

Thanks for noticing it. Yes, that's a mistake. The intended behavior was to configure it to timeout after 120 seconds.

> Can we set a timeout at runtime?

Yes, this should be a runtime configuration value.
